lib/Semantics/resolve-names.cpp

```diff
@@ -6056,6 +6058,21 @@ bool ResolveNamesVisitor::Pre(const parser::DefinedOpName &x) {
   return false;
 }
 
+void ResolveNamesVisitor::Post(const parser::AssignStmt &x) {
+  if (auto *name{ResolveName(std::get<parser::Name>(x.t))}) {
+    if (Symbol * symbol{name->symbol}) {
+      ConvertToObjectEntity(*symbol);
+    }
```

Comment:
I believe `ResolveName` doesn't return a name unless it's resolved, so these three lines could be replaced by:
```
ConvertToObjectEntity(DEREF(name->symbol));
```